### ISSUE TITLE
False positive when defining post-increment or post-decrement function in cpp file

### ISSUE THREAD

**Issue by armandas** [cpplint/cpplint#194]
The following causes a false positive `Using C-style cast.  Use static_cast<int>(...) instead  [readability/casting] [4]` error.

Test.hpp

```c++
struct Test
{
    Test operator++(int);
};
```

Test.cpp

```c++
#include "Test.hpp"

Test Test::operator++(int)
{
    return *this;
}
```

The issue is caused by the assumption that the `operator` keyword will always be prefixed with a space:

https://github.com/cpplint/cpplint/blob/35f28361aba21d3230b04241eeedc9ccff936632/cpplint.py#L5853

I propose that the condition is changed to the following:

```python
  if (context.endswith(' operator++') or context.endswith(' operator--') or
      context.endswith('::operator++') or context.endswith('::operator--')):
```
